hip-config.cmake.in

```diff
@@ -207,10 +206,10 @@ if(HIP_COMPILER STREQUAL "clang")
 
   # Add support for __fp16 and _Float16, explicitly link with compiler-rt
   set_property(TARGET hip::host APPEND PROPERTY
-    INTERFACE_LINK_LIBRARIES -L${HIP_CLANG_INCLUDE_PATH}/../lib/linux -lclang_rt.builtins-x86_64
+    INTERFACE_LINK_LIBRARIES "$<$<LINK_LANGUAGE:CXX>:${HIP_CLANG_INCLUDE_PATH}/lib/linux/libclang_rt.builtins-x86_64.a>"
```

Comment:
Doesn't this need to be `${HIP_CLANG_INCLUDE_PATH}/../lib` and not `${HIP_CLANG_INCLUDE_PATH}/lib`?

Reply:
@tomsang Are you able to make this change to the PR before merging?

Reply:
ping @tomsang this is a blocker for a set of Exascale apps for OLCF's Spock/Frontier.

Reply:
I'm working on this.   But maybe I need you help test after it's done.

Reply:
Sounds great, we could share code on ECP Tulip or OLCF Spock, if you have access to that.

Reply:
I cannot fix the issue under Robert account,    so I created  https://github.com/ROCm-Developer-Tools/HIP/pull/2280/  to cover the fix .
You may install the stack in terms of https://github.com/ROCm-Developer-Tools/HIP.

Reply:
Hi @tomsang, I created a reproducer for you under
https://github.com/ax3l/hip_cxxf_cmake

Simply build with:
```
CXX=$(which clang++) FC=$(which gfortran) cmake -S . -B build
cmake --build build
```

The error that you see is:
```
gfortran: error: unrecognized command line option ‘-mllvm’
clang-12: warning: argument unused during compilation: '-amdgpu-function-calls=false' [-Wunused-command-line-argument]
gfortran: error: unrecognized command line option ‘-amdgpu-early-inline-all=true’
gfortran: error: unrecognized command line option ‘-amdgpu-function-calls=false’
gfortran: error: unrecognized command line option ‘--offload-arch=gfx900’; did you mean ‘--offload-abi=ilp32’?
gfortran: error: unrecognized command line option ‘--offload-arch=gfx906’; did you mean ‘--offload-abi=ilp32’?
gfortran: error: unrecognized command line option ‘--offload-arch=gfx908’; did you mean ‘--offload-abi=ilp32’?
```